packages/ocs/dashboards/persistent-internal/capacity-trend-card/capacity-trend-card.tsx

```diff
@@ -60,36 +120,44 @@ const CapacityTrendCard: React.FC = () => {
       basePath: usePrometheusBasePath(),
     });
 
-  const [uptimeDay, uptimeDayError, uptimeDayLoading] = useCustomPrometheusPoll(
-    {
-      query:
-        CAPACITY_TREND_QUERIES(ocsCluster)[StorageDashboardQuery.UPTIME_DAYS],
-      endpoint: 'api/v1/query' as any,
-      basePath: usePrometheusBasePath(),
-    }
-  );
-  const uptimeDayMetric = parser(uptimeDay);
-
   const [totalUtil, totalUtilError, totalUtilLoading] = useCustomPrometheusPoll(
     {
-      query: CAPACITY_TREND_QUERIES(ocsCluster, uptimeDayMetric)[
+      query: CAPACITY_TREND_QUERIES(ocsCluster, retentionPeriod)[
         StorageDashboardQuery.UTILIZATION_VECTOR
       ],
       endpoint: 'api/v1/query' as any,
       basePath: usePrometheusBasePath(),
     }
   );
+  const [uptime, uptimeError, uptimeUtilLoading] = useCustomPrometheusPoll({
+    query: CAPACITY_TREND_QUERIES(ocsCluster, retentionPeriod)[
+      StorageDashboardQuery.UPTIME_DAYS
+    ],
+    endpoint: 'api/v1/query' as any,
+    basePath: usePrometheusBasePath(),
+  });
 
-  const loadError = availableCapacityError || uptimeDayError || totalUtilError;
+  const loadError = availableCapacityError || totalUtilError || uptimeError;
   const loading =
-    availableCapacityLoading || uptimeDayLoading || totalUtilLoading;
+    availableCapacityLoading || totalUtilLoading || uptimeUtilLoading;
+
+  let daysUp: number;
+  if (!loading && !loadError) {
+    daysUp = calculateDaysUp(uptime);
+  }
 
   const totalUtilMetric = parser(totalUtil);
-  const avgUtilMetric = totalUtilMetric / uptimeDayMetric;
+  const avgUtilMetric =
+    !!retentionPeriod && !!totalUtilMetric
+      ? totalUtilMetric / Math.ceil(daysUp) // do a ceil function, if it's a new cluster (dayUp < 1)
```

Comment:
`daysUp` can be `null` or `undefined`, in which case `ceil` will return `0` or `NaN` respectively...
```suggestion
    !!daysUp && !!totalUtilMetric
      ? totalUtilMetric / Math.ceil(daysUp) // do a ceil function, if it's a new cluster (dayUp < 1)
```

Reply:
fixed